miden-lib/asm/miden/kernels/tx/prologue.masm

```diff
@@ -495,56 +495,64 @@ end
 #! - authenticate the note and its metadata in the note Merkle tree from the block the note was
 #!   created in.
 #!
-#! Operand stack: [AUTH_DIGEST]
-#! Advice stack: [leaf_pos, SUB_HASH, NOTE_ROOT, note_index]
+#! Operand stack: [NOTE_HASH]
+#! Advice stack: [leaf_pos, BLOCK_SUB_HASH, NOTE_ROOT, note_index]
 #! Output: []
 #!
 #! Where:
-#! - AUTH_DIGEST is the digest of the input note data computed as hash(NOTE_HASH, NOTE_METADATA).
+#! - NOTE_HASH is the digest of the input note data computed as hash(NOTE_HASH, NOTE_METADATA).
 #! - leaf_pos is the position of the leaf in the MMR associated with the block the note was created.
 #!   in. This is equivalent to the block number.
-#! - SUB_HASH, the block's sub_hash for which the note was created.
+#! - BLOCK_SUB_HASH, the block's sub_hash for which the note was created.
 #! - NOTE_ROOT, the merkle root of the note's tree.
 #! - note_index is the index of the note in the note Merkle tree.
 proc.authenticate_note.2
-    # load data required for MMR get operation
+    # Load the BLOCK_HASH from the CHAIN_MMR
+    # ---------------------------------------------------------------------------------------------
+
     exec.memory::get_chain_mmr_ptr adv_push.1
-    # => [leaf_pos, chain_mmr_ptr, AUTH_DIGEST]
+    # => [leaf_pos, chain_mmr_ptr, NOTE_HASH]
 
-    # get the chain MMR leaf associated with the block the note was created in
     exec.mmr::get
-    # => [MMR_LEAF, AUTH_DIGEST]
+    # => [BLOCK_HASH, NOTE_HASH]
 
-    # prepare the stack to read the sub hash and note root from the advice provider
-    locaddr.0 padw padw padw
-    # => [PAD, PAD, PAD, mem_ptr, MMR_LEAF, AUTH_DIGEST]
+    locaddr.0
+    # => [mem_ptr, BLOCK_HASH, NOTE_HASH]
 
-    # read the core hash and note root from the advice provider
+    # Load and authenticate the NOTE_ROOT
+    # ---------------------------------------------------------------------------------------------
+
+    # read data from advice and compute hash(BLOCK_SUB_HASH || NOTE_ROOT)
+    padw padw padw
     adv_pipe hperm
-    # => [PERM, PERM, PERM, mem_ptr', MMR_LEAF, AUTH_DIGEST]
+    # => [PERM, COMPUTED_BLOCK_HASH, PERM, mem_ptr', BLOCK_HASH, NOTE_HASH]
 
-    # extract the digest and assert it matches MMR_LEAF
-    dropw movup.8 drop movupw.2
-    assert_eqw.err=ERR_PROLOGUE_NOTE_MMR_DIGEST_MISMATCH
-    # => [AUTH_DIGEST]
```

Comment:
This is the motivation to open the PR.

The comment about the stack was incorrect, and it really threw me off. At this stage the actual state is:

```
# => [PERM, AUTH_DIGEST]
```

The `loc_loadw.1` will then overwrite the `PERM` value with the `NOTE_ROOT` saved in the procedure's local memory.